components/inputs/input-text.js

```diff
@@ -150,5 +249,32 @@ class InputText extends RtlMixin(LitElement) {
 		e.preventDefault();
 	}
 
+	_handleMouseEnter() {
+		this._hovered = true;
+	}
+
+	_handleMouseLeave() {
+		this._hovered = false;
+	}
+
+	_onSlotChange(e) {
+		const children = e.target.assignedNodes();
+		if (children.length > 0) {
+			const slotName = e.target.name;
+			const affectedSlotName = this.dir !== 'rtl' ? slotName : (slotName === 'left' ? 'right' : 'left');
+			const slotContent = children[0];
+			const slotContentWidth = slotContent.offsetWidth;
+			const diff = (this.slotPaddingWidth * documentFontSize - slotContentWidth) / 2;
```

Comment:
Could you leverage the `calc` above instead of needing to use the document font size? Like here we could just set `this._slotLeftWidth = e.target.offsetWidth;` and then when it renders do a `calc(0.75rem + ${this._slotLeftWidth}px)`. Would need some tweaking.

Reply:
The problem is that `left` or `right` is being set on the slotcontent so it makes using a `styleMap` and using `calc` in `render more complicated/maybe not possible. I'd like to simplify this though so I'm looking into workarounds.